### PR TITLE
add go library

### PR DESCRIPTION
**Pull request by cristianciutea**
This PR adds:
- golang module implementing thrift interface to call nrjmx java tool
- check that pushed generated files are correct
- update tests to use the new module instead of nr-sdk

### CODE REVIEW

File: build/build.mk [hunk 1]
```diff
@@ -1,14 +1,40 @@
...
+
+.PHONY : code-gen-deps
+code-gen-deps:
+		@($(DOCKER_BIN) build -t nrjmx-code-generator ./commons/.)
+
+.PHONY : code-gen-deps, code-gen
+code-gen: 
+	@($(DOCKER_THRIFT) thrift -r --out src/main/java/ --gen java ./commons/nrjmx.thrift)
+	@($(DOCKER_THRIFT) thrift -r --out gojmx/ --gen go:package_prefix=github.com/newrelic/nrjmx/gojmx/,package=nrprotocol ./commons/nrjmx.thrift)
+
```
Comment (rubenruizdegauna) [inline build/build.mk:30]: I tried `make code-gen` and failed. shouldn't this be:
```
.PHONY : code-gen-deps
code-gen-deps:
		@($(DOCKER_BIN) build -t nrjmx-code-generator ./commons/.)

.PHONY : code-gen
code-gen: code-gen-deps
	@($(DOCKER_THRIFT) thrift -r --out src/main/java/ --gen java ./commons/nrjmx.thrift)
	@($(DOCKER_THRIFT) thrift -r --out gojmx/ --gen go:package_prefix=github.com/newrelic/nrjmx/gojmx/,package=nrprotocol ./commons/nrjmx.thrift)
```

File: commons/nrjmx.thrift [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  3: i32 port,
+  4: string uriPath,
+  5: string username,
+  6: string password,
+  7: string keyStore,
+  8: string keyStorePassword,
+  9: string trustStore,
+  10: string trustStorePassword,
+  11: bool isRemote,
+  12: bool isJBossStandaloneMode
```
Comment (rubenruizdegauna) [inline commons/nrjmx.thrift:15]: would it make sense to have `isStandaloneMode` instead o jboss related one? os is it only jboss related?

File: gojmx/process.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+var defaultNrjmxExec = "/usr/local/bin/nrjmx"
+
+func getNrjmxExec() string {
+	if os.Getenv("NR_JMX_TOOL") != "" {
+		return os.Getenv("NR_JMX_TOOL")
+	}
+	return defaultNrjmxExec
+}
+
+// var defaultNrjmxExec = "/home/cristi/workspace/cppc/java/nrjmx/bin/nrjmx"
```
Comment (rubenruizdegauna) [inline gojmx/process.go:24]: ☝️ 😄

File: gojmx/process.go [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+			if err != nil {
+				if _, isAlreadyClosed := err.(*os.PathError); !isAlreadyClosed && err != io.EOF {
+					fmt.Fprintf(os.Stderr, "error while reading stderr: '%v'", err)
+					continue
+				}
+				return
+			}
+			fmt.Fprint(os.Stderr, line)
+		}
+	}()
```
Comment (rubenruizdegauna) [inline gojmx/process.go:77]: why is this done here and not in the jmxProcess?

File: commons/nrjmx.thrift (hunk 2 above)
Comment (cristianciutea) [reply commons/nrjmx.thrift:15]: is only jboss related

**Review by rubenruizdegauna — APPROVED** [review]
LGTM

**Review by rogercoll — APPROVED** [review]

File: build/build.mk (hunk 1 above)
Comment (cristianciutea) [reply build/build.mk:30]: tackling this in a new PR

**Pull request merged**